### PR TITLE
Resize icon based on calendar event size

### LINKED ISSUE #6973: Calendar events with short durations display the calendar icon even if there's no room for it
The title says it all.
Here's an example: 
image

### PR DESCRIPTION
**Pull request by heytherewill**
## What's this?
This resizes the calendar icon so it always looks good no matter the size of the event

### Relationships
Closes #6973

## How is it done?
I'm simply resizing the bitmap when needed, no big deal

### Why not another way?
I'm currently centering the bitmap when it gets resized. I could align it to the left as well, I just need @maritoggl to say the word. 

Here's what each of them look like (notice the icon of the `Guy ghj` event)

| Centered (current behavior)| Left aligned|
| ------------- | ------------- |
|image|image

### Side effects
None

## :squid: Permissions
Anyone, but not after @maritoggl has stated what the correct behavior is

### CODE REVIEW

File: Toggl.Droid/Views/Calendar/CalendarDayView.Events.cs [hunk 1]
```diff
@@ -252,8 +252,23 @@ private void drawCalendarEventItemShape(Canvas canvas, CalendarItem item, RectF
...
+        }
+
+        private RectF calculateCalendarIconRect(RectF calendarItemRect)
+        {
+            var containerHeight = calendarItemRect.Height();
+            var scale = containerHeight > calendarIconBitmap.Height ? 1.0f : containerHeight / calendarIconBitmap.Height;
+
+            var width = calendarIconBitmap.Width * scale;
+            var top = calendarItemRect.Top;
+            var left = calendarItemRect.Left + ((calendarIconBitmap.Width - width) / 2);
```
Comment (daividssilverio) [inline Toggl.Droid/Views/Calendar/CalendarDayView.Events.cs]: The bitmap left doesn't seem to be calculated properly.
image

File: Toggl.Droid/Views/Calendar/CalendarDayView.Events.cs [hunk 2]
```diff
@@ -252,8 +252,23 @@ private void drawCalendarEventItemShape(Canvas canvas, CalendarItem item, RectF
...
+            var right = left + width;
+            var bottom = top + calendarIconBitmap.Height * scale;
+
+            return new RectF(left, top, right, bottom);
```
Comment (daividssilverio) [inline Toggl.Droid/Views/Calendar/CalendarDayView.Events.cs]: Please, don't create rects, have one field for it and reuse it.

**Comment by maritoggl** [comment]
Align

File: Toggl.Droid/Views/Calendar/CalendarDayView.Events.cs (hunk 1 above)
Comment (heytherewill) [reply Toggl.Droid/Views/Calendar/CalendarDayView.Events.cs]: It's centered, but, as Mari said below, they should be left-aligned instead

**Comment by daividssilverio** [comment]
🦑 it, that is.

**Pull request merged**